pkg/ddevapp/config_test.go

```diff
@@ -607,12 +608,41 @@ func TestConfigValidate(t *testing.T) {
 
 	appName := app.Name
 	appType := app.Type
+	ddevVersion := versionconstants.DdevVersion
 
 	err = app.ValidateConfig()
 	if err != nil {
 		t.Fatalf("Failed to app.ValidateConfig(), err=%v", err)
 	}
 
+	app.DdevVersionConstraint = ">= 1.twentythree"
```

Comment:
Could you please explore making these constraints more flexible? Otherwise we'll deal with needless breakage over the years going forward.

Reply:
I didn't understand this. What do you mean? The constrains format comes straight from semver.

Reply:
On this test I made it so that I fails.

Reply:
Sorry, I was referring to the whole of the section. Please make it so that this test doesn't have to be edited when v1.23.0 comes out, for example.

Reply:
It doesn't. I am also setting a specific ddev version so that tit will always fail.

Reply:
The first one will fail because it's just invalid

Reply:
I'd like to see `v1.22.3-11-g8baef014e` being tested since that's what I have in `DDEV_VERSION` when I compile from source.
For example `v1.22.3-11-g8baef014e >= 1.22`, `v1.22.3-11-g8baef014e < 1.23`

Reply:
@jonaseberle ok I added more tests. How do you get that version? I am getting `22be588e1-dirty` as version on my compiled `ddev`.

A warning here, `v1.22.3-11-g8baef014e` is considered a pre-release, so `v1.22.3-11-g8baef014e >= 1.22` will actually fail as the constraint, as it's defined by the semver module, excludes pre-releases by default. It will fail with any constraint that doesn't include pre-releases. `v1.22.3-11-g8baef014e >= 1.0` will also fail.

If you want to include pre-releases, you have to use a constraint that includes them:

`v1.22.3-11-g8baef014e >= 1.22.0-0` will pass.

Both things are tested.

https://github.com/Masterminds/semver#working-with-prerelease-versions

I will try documenting this.

Reply:
That's the VERSION we get in Makefile by `git describe --tags --always --dirty` when there are tags somewhere up in your commit history (and in my case nothing uncommitted, so not `-dirty`). IMHO we should make that format supported somehow.

Reply:
`dirty` means that something isn't committed (`git status` shows uncommitted files)

Reply:
`v1.22.3-11-g8baef014e` and `v1.22.3-11-g8baef014e-dirty` should already work with the semver comparisons if they're working right and being used right.

Reply:
As I am in my fork I don't have the tags so I don't get that. Ok, they work as long as you have a constrain that includes them. I am not if there's an actionable thing to do here, and this is very specific for dev verisions.

`v1.22.3-11-g8baef014e` with a constraint of `>= 1.22` will not allow ddev to run
`v1.22.3-11-g8baef014e` with a constraint of `>= 1.22.0-0` will.

If you would like `v1.22.3-11-g8baef014e` with a constraint of `>= 1.22` to be "allowed" I am not sure what to do there if we are going to use this version, which I think it's the version we should use.

If there's something specific for me to do around this I am happy to do it, if there's not and it's OK as it is, let's resolve it.

Reply:
I don't consider this a bad thing but I see how this can confuse users.

One alternative would be to strip pre-releases out of the version

Reply:
that would make any `v1.23.0-whatever` be `>=1.23`

Reply:
If this is desirable I can work on it

Reply:
I think this is the way to go actually

Reply:
the only problem there is that 

`v1.23.0-alpha1` with constrain of `>= v1.23.0-alpha2` will still work if I remove`-alpha1` from the version

🤔

Reply:
If you wish to play with it: https://go.dev/play/p/9-OII74ev21

Reply:
I think this is ok and can be documented, if a user is using an alpha/dev version he's advanced enough, but otherwise I'll wait for guidance.

Reply:
I am not quite sure if that means leaving the way it is or not :)

Reply:
>  v1.22.3-11-g8baef014e with a constraint of >= 1.22.0-0 will
I think that's the way to go then. Looks good from my side feature-wise.